src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -185,75 +195,483 @@ uint32 insert(struct fsobj *o, struct fsobj *root)
+        buffer                          += bytes_remaining;
+        bytes_to_write                  -= bytes_remaining;
+        o->position.part_offset         = 0;
+        if (o->position.open_part->next == NULL) {
+            struct f_part *part;
+            newpart_get(&part);
+            part->memtype               = DYNAMIC;
+            part->file                  = o;
+            part->next                  = NULL;
+            part->prev                  = o->position.open_part;
+            o->position.open_part->next = part;
+        }
+
+        o->position.open_part           = o->position.open_part->next;
+        o->position.addr                = o->position.open_part->data;
+        bytes_remaining                 = F_PART_DATA_SIZE - o->position.part_offset;
+    }
+
+    //bytes_to_write < bytes_remaining
+    memcpy(o->position.addr, buffer, bytes_to_write);
+    o->position.addr                    += bytes_to_write;
+    o->position.part_offset             += bytes_to_write;
+    o->position.file_offset             += bytes_to_write;
+    if (o->size < o->position.file_offset) {
+        o->size                         = o->position.file_offset;
+    }
     return nbytes;
 }
 
-struct fsobj *find(const char *path, struct fsobj *root)
+void file_rewinddir()
 {
-    assert(root);
-    assert(strlen(root->name) < strlen(path));
-    while (1) {
-        assert(strcmp(root->name, path) !=0 );
-        assert(strspn(path, root->name) >= strlen(root->name));
-        if (!root->child) return NULL;
+    if (!openfile) return;
+    if (openfile->parent == NULL) return;
+    assert(openfile->parent->child);
+    openfile = openfile->parent->child;
+    return;
+}
 
-        root = root->child;
+struct fsobj *file_find(const char *path)
+{
+    //paths should always begin with '/' dir names do not
+    if (path[0] != '/') return NULL;
+    path++;
+
+    assert(path);
+    if(!openfs || !openfs->root) return NULL;
+    struct fsobj *root = openfs->root;
+    assert(root && root->name);
+    if (strlen(root->name) > strlen(path)) return NULL;
+
+    while (1) {
 
+        // iterate through linked list of children
         while (strspn(path, root->name) < strlen(root->name)) {
             if (!strcmp(root->name, path)) return root;
             if (!root->next) return NULL;
             root = root->next;
         }
 
+        //root must now be parent of final location
+        if (!strcmp(root->name, path)) return root;
+        assert(strspn(path, root->name) >= strlen(root->name));
+        if (!root->child) return NULL;
+
+        root = root->child;
+        while (path[0] != '/') {
+            path++;
+        }
+        path++;
+
     }
     PANIC("Unreachable Statement");
     return NULL;
 }
 
+int32 file_mkdir(const char *path, uint32 access)
+{
+    assert(path);
+    struct fsobj *o;
+    if (newfile_get(&o)) return OS_FS_ERR_DRIVE_NOT_CREATED;
+    o->name = path_to_name((char *) path);
```

Comment:
You're defining both these functions, you should never need to cast away const...